query/iterator.gen.go.tmpl

```diff
@@ -410,11 +411,75 @@ func (itr *{{$k.name}}SortedMergeIterator) pop() (*{{$k.Name}}Point, error) {
 type {{$k.name}}SortedMergeHeap struct {
 	opt   IteratorOptions
 	items []*{{$k.name}}SortedMergeHeapItem
+	// if each input comes from a unique single time series, we can make a shortcut.
+	// detection of the shortcut introduces some overhead but it gets significant
+	// performance improvement in cases like SELECT * FROM m GROUP BY *
+	fast bool
+}
+
+func (h *{{$k.name}}SortedMergeHeap) detectFast() {
+       for _, item := range h.items {
+               if item.itr.Stats().SeriesN != 1 {
+                       return
+               }
+       }
+
+       hasDup := false
+       s := make([]*{{$k.name}}SortedMergeHeapItem, len(h.items))
+       copy(s, h.items)
+
+       less := func(i, j int) bool {
+               x, y := s[i].point, s[j].point
+               ret := strings.Compare(x.Name, x.Name)
+               if ret == +1 {
```

Comment:
Please change all these instances of `+1` to `1`.

Reply:
solved now